docs/book/v3/features/helpers/url-helper.md

```diff
@@ -70,6 +70,51 @@ Each method will raise an exception if:
 > mezzio/mezzio-helpers. Prior to that version, the helper only
 > accepted the route name and route parameters.
 
+### Reusing Matched Route Result Parameters
+
+When you're on a route that has many parameters, often times it makes sense to reuse
+currently matched route parameters instead of assigning them explicitly, this is the 
+default behaviour.
+
+As an example, we will imagine being on a detail page for our `blog` route. We want
+to display links to the `edit` and `delete` actions without having to assign the ID
+again:
+
+```php
+// Current URL: /blog/view/777
+
+$this->url('blog', ['action' => 'edit']);
+$this->url('blog', ['action' => 'delete']);
```

Comment:
The route parameter `action` looks like handling of multiple routes in a single middleware class, but this is not the typical usage in Mezzio or what the documentation describes. Therefore, it should be omitted.

Reply:
Good point, will rewrite.